Gem/Code/Source/Components/Multiplayer/EnergyBallComponent.cpp

```diff
@@ -5,120 +5,142 @@
-    void EnergyBallComponentController::HandleRPC_LaunchBall([[maybe_unused]] AzNetworking::IConnection* invokingConnection,
-        const AZ::Vector3& startingPosition, const AZ::Vector3& direction)
+    void EnergyBallComponentController::HandleRPC_LaunchBall([[maybe_unused]] AzNetworking::IConnection* invokingConnection, const AZ::Vector3& startingPosition, const AZ::Vector3& direction, const Multiplayer::NetEntityId& owningNetEntityId)
     {
-        m_direction = direction;
+        HideEnergyBall();
 
-        using RigidBodyBus = Physics::RigidBodyRequestBus;
-        RigidBodyBus::Event(GetEntityId(), &RigidBodyBus::Events::DisablePhysics);
+        m_filteredNetEntityIds.clear();
+        m_filteredNetEntityIds.insert(owningNetEntityId);
+        m_filteredNetEntityIds.insert(GetNetEntityId());
+        m_direction = direction;
 
         // move self and increment resetCount to prevent transform interpolation
         AZ::TransformBus::Event(GetEntityId(), &AZ::TransformBus::Events::SetWorldTranslation, startingPosition);
         GetNetworkTransformComponentController()->ModifyResetCount()++;
 
-        RigidBodyBus::Event(GetEntityId(), &RigidBodyBus::Events::EnablePhysics);
-        RigidBodyBus::Event(GetEntityId(), &RigidBodyBus::Events::SetLinearVelocity, direction * aznumeric_cast<float>(m_speed));
-
-        GameplayEffectsNotificationBus::Broadcast(&GameplayEffectsNotificationBus::Events::OnPositionalEffect,
-            SoundEffect::EnergyBallTrapProjectile, GetEntity()->GetTransform()->GetWorldTranslation());
+        Physics::RigidBodyRequestBus::Event(GetEntityId(), &Physics::RigidBodyRequestBus::Events::EnablePhysics);
+        Physics::RigidBodyRequestBus::Event(GetEntityId(), &Physics::RigidBodyRequestBus::Events::SetLinearVelocity, direction * GetGatherParams().m_travelSpeed);
     }
-#endif
 
-    void EnergyBallComponentController::HideEnergyBall()
+    void EnergyBallComponentController::CheckForCollisions()
     {
-        Physics::RigidBodyRequestBus::Event(GetEntityId(), &Physics::RigidBodyRequestBus::Events::DisablePhysics);
-
-        // move self and increment resetCount to prevent transform interpolation
-        AZ::TransformBus::Event(GetEntityId(), &AZ::TransformBus::Events::SetWorldTranslation, AZ::Vector3::CreateAxisZ(-1000.f));
-        GetNetworkTransformComponentController()->ModifyResetCount()++;
-    }
+        const AZ::Vector3& position = GetEntity()->GetTransform()->GetWorldTM().GetTranslation();
+        const HitEffect& effect = GetHitEffect();
 
-    void EnergyBallComponentController::OnCollisionBegin(const AzPhysics::CollisionEvent& collisionEvent)
-    {
-        if (collisionEvent.m_body2)
+        IntersectResults results;
+        const ActivateEvent activateEvent{ GetEntity()->GetTransform()->GetWorldTM(), position, Multiplayer::InvalidNetEntityId, Multiplayer::InvalidNetEntityId };
+        GatherEntities(GetGatherParams(), activateEvent, m_filteredNetEntityIds, results);
```

Comment:
it's probably unlikely, but if this is not doing a shape cast from the old to new position, it's possible a player might run at an energy ball and it not hit depending on the player speed and energy ball speed.  so long as the ball and player move slow enough we should be good.

Reply:
Yeah, tunneling could definitely be a potential issue, but the ball does move quite slowly typically. I've thrown a video of what I've configured into the PR description, including the revised jump-pad effect, but design can tune this even lower if they want players to try more skill-based jumping and dodging of these traps.